articles/validating-xml-using-xsd/index.md

```diff
@@ -0,0 +1,262 @@
+
+### What is XSD?
+> According to Wikipedia, [XML Schema Definition (XSD)](https://en.wikipedia.org/wiki/XML_Schema_(W3C)) can be used to express a set of rules to which an XML document must conform to be considered "valid" according to that schema. However, unlike most other schema languages, XSD was also designed with the intent that the determination of a document's validity would produce a collection of information adhering to specific data types.
+
+You may be wondering what is the need for XSD when we already have DTD. XSD can validate much better than DTD, in terms of constraints verification, usage, and relationships between elements and its attributes. According to [W3C](https://www.w3.org/TR/xmlschema11-1/), XSD defines, describes, and catalogs XML vocabularies for classes of XML documents.
+
+### Step by Step guide for validation
+#### XML document
+As you know, XSD is used for complex validation of XML document. So, let's take up some complex XML document for validating. By validating it, understanding of concepts would be much better. In this article, we are not going to cover the basics of building XML document. To keep it simple, let's validate the below XML code. Full XML code can be found [here](https://gist.github.com/srishilesh/5d028a5d9acef3818b014cd4595c9ebe).
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<root xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="validation_xsd.xsd">
+    <event>
+        <event_id>1</event_id>
+        <event_name>ICPC</event_name>
+        <event_desc>5 coding questions in 2 hrs</event_desc>
+        <event_tags>
+            <tag id="1" displayName="coding">coding</tag>
+        </event_tags>
+        <event_type_participation>Team</event_type_participation>
+        <event_timing>
+            <event_start_datetime>2020-04-01T10:00:00</event_start_datetime>
+            <event_end_datetime>2020-04-01T12:00:00</event_end_datetime>
+        </event_timing>
+        <event_organizer_email>abc@gmail.com</event_organizer_email>
+        <event_organizer_phone>1234567890</event_organizer_phone>
+    </event>
+</root>
+```
+
+#### Understanding XSD terminologies
+##### Elements
+A simple element in XSD contains the text of various data types like boolean, string, date, etc. or any custom type. Restrictions can be added to it, to maintain consistency of values. A sample XSD element can be defined as:
+
+Syntax:
+```xml
+<xs:element name="xxx" type="yyy"/>
+```
+where `type` can be of
+- `xs:string`
+- `xs:integer`
+- `xs:decimal`
+- `xs:date`
+- `xs:time`
+- `xs:boolean`
+
+##### Attributes
+A complex element with element-specific values is known as attributes. It is not mandatory to mention attributes in elements. When attributes are specified, an element is known to be a complex element.
```

Comment:
> A complex element with element-specific values is known as attributes

"A complex element with element-specific values is known as an attribute." is the correct form of this sentence.